python/lib/client/dmod/client/request_clients.py

```diff
@@ -264,9 +266,27 @@ class DatasetExternalClient(DatasetClient,
     """
 
     # In particular needs - endpoint_uri: str, ssl_directory: Path
-    def __init__(self, *args, **kwargs):
+    def __init__(self, *args, cache_session: bool = True, **kwargs):
         super().__init__(*args, **kwargs)
-        self._cached_session_file = Path.home().joinpath('.dmod_client_session')
+        self._cached_session_file: Optional[Path] = (
+            Path.home().joinpath(".dmod_client_session") if cache_session else None
+        )
+
+    @classmethod
+    def from_session(cls, *, endpoint_uri: str, ssl_directory: Path, session: Session, cache_session: bool = True, **kwargs) -> Self:
+        """
+        Create a `DatasetExternalClient` from an existing `Session` instance.
+
+        Note, the passed `Session` object will not be written to disk even if the `cache_session`
+        flag is present.  However, if the `Session` instance expires and a new session is acquired,
+        it will be cached if `cache_session` is set.
```

Comment:
> Note, the passed `Session` object will not be written to disk even if the `cache_session` flag is present.

Took me a little to figure out why this was.  Seems to be another artifact of not reusing existing, inherited functionality very well.  I suspect we will need to address that.